cli/src/tests/query_test.rs

```diff
@@ -737,6 +737,57 @@ fn test_query_matches_capturing_error_nodes() {
     });
 }
 
+#[test]
+fn test_query_matches_capturing_missing_nodes() {
+    allocations::record(|| {
+        let language = get_language("javascript");
+        let query = Query::new(
+            &language,
+            "
+            (MISSING) @missing-node
+            ",
+        )
+        .unwrap();
+
+        // Missing anonymous nodes
+        assert_query_matches(
+            &language,
+            &query,
+            "if ({a: 'b'} {c: 'd'}) {
+  x = function(a) { b; } function(c) { d; }
+}",
```

Comment:
Can we comment what the missing node is and where with a carat? It's not immediately obvious to anyone else *what* exactly is missing in this test. Also, try to minimize the test case, there's too much going on that is not needed to test that querying a missing node works.